pkg/cloudprovider/types.go

```diff
@@ -75,14 +78,32 @@ type InstanceType interface {
 	// Overhead is the amount of resource overhead expected to be used by kubelet and any other system daemons outside
 	// of Kubernetes.
 	Overhead() v1.ResourceList
-	// Price is a metric that is used to optimize pod placement onto nodes as well as determining if replacing a node with
-	// a cheaper instance type is possible when performing consolidation.
-	Price() float64
 }
 
 // An Offering describes where an InstanceType is available to be used, with the expectation that its properties
 // may be tightly coupled (e.g. the availability of an instance type in some zone is scoped to a capacity type)
 type Offering struct {
 	CapacityType string
 	Zone         string
+	Price        float64
+	Available    bool
+}
+
+// AvailableOfferings filters the offerings on the passed instance type
+// and returns the offerings marked as available
+func AvailableOfferings(it InstanceType) []Offering {
+	return lo.Filter(it.Offerings(), func(o Offering, _ int) bool {
+		return o.Available
+	})
+}
+
+// GetOffering gets the offering from passed offerings that matches the
+// passed zone and capacity type
+func GetOffering(ofs []Offering, ct, zone string) (Offering, error) {
```

Comment:
nit: maybe slightly more intuitive to hang this off of instanceType, instead of []Offering.

Reply:
nit: I also noticed that you're never actually consuming the error as an error, you're just consuming it as a boolean. Consider returning `Offering, bool`, which also happens to be an identical interface to `lo.Find`, which can simplify your code/concepts.